### ISSUE TITLE
[SDL2]Help items after DOS commands cannot be reached with CGA/EGA/Tandy/PCJr rendering mode.

### ISSUE THREAD

**Issue by FredBezies** [joncampbell123/dosbox-x#2631]
**Describe the bug**
Hello.

I noticed that in Dosbox-X 0.83.15, if you're using CGA/EGA/Tandy/PCJr, you cannot access to help items under DOS Commands.

Commands list it too wide to let mouse cursor goes under it.

**To Reproduce**
1. Grab Dosbox-X 0.83.15
2. In Configuration Tool, go to main and choose  CGA/EGA/Tandy/PCJr rendering mode
3. Save and restart with the new configuration file.
4. Go to help menu and click on it. Go down to DOS Commands.

**Expected behavior**
DOS Commands box to be a little on the right to get access to other menu items.

**Screenshots**

![Capture d’écran de 2021-07-02 11-10-29](https://user-images.githubusercontent.com/8571419/124251403-28855900-db26-11eb-8c3f-0b08bd653827.png)

**Environment (please complete the following information):**
 - Archlinux
 - DOSBox-X 0.83.15 built with SDL2.
 - Only modified main/machine to use  CGA/EGA/Tandy/PCJr rendering mode instead of svga_s3

**Comment by FredBezies** [comment]
Here is the result with DosBox using SDL1.

![Capture d’écran de 2021-07-02 11-19-54](https://user-images.githubusercontent.com/8571419/124252712-87979d80-db27-11eb-96ea-1a121458cd71.png)

So something is broken with SDL2 and help menu items.

**Comment by grapeli** [comment]
I do not note anything like that. Rather, it is a matter of window manager behavior.
Though I don't know why this window goes full screen and back to its original size.

https://user-images.githubusercontent.com/452325/124260944-c590c380-db1f-11eb-917d-3d0705e48fcc.mp4

It works badly under the wayland. Due to a malfunctioning fullscreen (zoom 100x).
It does not return to its original size by itself.
https://user-images.githubusercontent.com/452325/124260501-48fde500-db1f-11eb-8c88-947eb503252e.mp4

**Comment by grapeli** [comment]
It works badly under the wayland. Due to a malfunctioning fullscreen (zoom 100x).
It does not return to its original size by itself.

https://user-images.githubusercontent.com/452325/124261325-2c15e180-db20-11eb-8dca-3abf51472cf7.mp4

**Comment by FredBezies** [comment]
I noticed this bug only with CGA/EGA/Tandy/PCJr rendering mode. When I go back to svga_s3, bug is dead.

I tried replacing --enable-debug by --disable-debug in configure line.

Here is the result:

![Capture d’écran de 2021-07-02 13-05-02](https://user-images.githubusercontent.com/8571419/124265634-3d69e880-db36-11eb-87eb-90ea45c562c9.png)

Looks like Debug menu is guilty here. Modifying bug title.

**Comment by grapeli** [comment]
With the additional "Debug" item in the menu, you can also access all items. A bit more difficult. However, they are achievable.

https://user-images.githubusercontent.com/452325/124271611-31c5f400-db2d-11eb-9ccd-1f8e5692af8c.mp4

**Comment by FredBezies** [comment]
> With the additional "Debug" item in the menu, you can also access all items. A bit more difficult. However, they are achievable.
> dosbox-x-sdl2-i3.mp4

This is a manual workaround :smile:

**Comment by grapeli** [comment]
Possible.

Run any game in 320x200 resolution from dosbox-x running like this.
`dosbox-x -set doublescan=false -set scaler=none`
You cannot access any menu items beyond Capture - Drive, [Debug], Help.

**Comment by Wengier** [comment]
I agree that the SDL-drawn menu can be sometimes inflexible with small Windows like in this case. It is possible to address this later, but for Linux SDL2 build, you can start the DOSBox-X window maximized with the setting `maximize=true`, and such issues won’t occur with bigger Windows.

**Comment by grapeli** [comment]
Unfortunately, the non-maximized window (SDL2) does not work completely properly.
It looks the worst under JSLinux.

https://user-images.githubusercontent.com/452325/124368094-f6bae200-dc4c-11eb-89c3-8d8bed8b13ee.mp4

Fully static dosbox version 0.83.14 (No fluidsynth, opengl, slirp.).
[dosbox-x-0.83.14.linux.static.tar.gz](https://github.com/joncampbell123/dosbox-x/files/6759115/dosbox-x-0.83.14.linux.static.tar.gz)
Under [JSLinux](https://bellard.org/jslinux/vm.html?url=alpine-x86-xwin.cfg&mem=256&graphic=1) you can run like this:
```
gzip -dc dosbox-x-0.83.14.linux.static.tar.gz | tar xf - -C/
dosbox-x
```
